img.js

```diff
@@ -415,6 +415,15 @@ class Image {
       metadata.height = width;
     }
 
+    if(metadata.pageHeight) {
+      // When the { animated: true } option is provided to sharp, animated
+      // image formats like gifs or webp will have an inaccurate `height` value
```

Comment:
Multi-page/frame images are processed by libvips as a "toilet roll" of images, which is why the height is the total of all frames, and therefore the page height is also made available. Hope this helps explain the situation.